pytorch_lightning/trainer/trainer.py

```diff
@@ -149,6 +149,7 @@ def __init__(
         auto_lr_find: Union[bool, str] = False,
         replace_sampler_ddp: bool = True,
         terminate_on_nan: bool = False,
+        detect_anomaly: bool = False,
```

Comment:
from https://github.com/PyTorchLightning/pytorch-lightning/issues/8313#issuecomment-894098307 , if we're adding this argument, let's deprecate `terminate_on_nan` in favor of `detect_anomaly`

Reply:
I do it by adding a deprecation warning when that parameter is used and suggest using "detect_anomaly". Right?
Also, which version is the current one and when will it be removed?

Reply:
> I do it by adding a deprecation warning when that parameter is used and suggest using "detect_anomaly". Right?

Yes: 
- make `terminate_on_nan` an `Optional[bool]` with default value = None
- if it's not None, log a deprecation warning and suggest to use `detect_anomaly`

This will be marked as deprecated in the v1.5 release and will be removed in the v1.7 release

you could use https://github.com/PyTorchLightning/pytorch-lightning/pull/8989 as a reference